Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +658,81 @@ class InAppMessageViewTest: IntegrationTest {
+
+        // errorWithMessage called
+        XCTAssertTrue(eventListenerMock.errorWithMessageCalled)
+        XCTAssertEqual(eventListenerMock.errorWithMessageCallsCount, 1)
+        XCTAssertEqual(eventListenerMock.errorWithMessageReceivedArguments, InAppMessage(gistMessage: givenInlineMessage))
+
+        // Other listeners not called
+        XCTAssertFalse(eventListenerMock.messageActionTakenCalled)
+        XCTAssertFalse(eventListenerMock.messageShownCalled)
+        XCTAssertFalse(eventListenerMock.messageDismissedCalled)
+    }
+
+    @MainActor
+    func test_givenAttemptToShowInlineMessageFails_expectMessageNotShown() async {
+        messagingInAppImplementation.setEventListener(eventListenerMock)
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // Inline message does not display
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+    }
+
+    @MainActor
+    func test_givenAttemptToShowInlineMessage_EventListenerNotSet_expectFailToShow_expectNoCallback() async {
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessageWithError(givenInlineMessage, insideOfInlineView: inlineView)
+
+        // Inline message does not display
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+
+        // errorWithMessage not called as eventListener is not set
+        XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
+    }
+
+    // messageActionTaken when close button is tapped
+    @MainActor
+    func test_givenShowInlineMessage_givenTapCloseButton_expectInlineToHide_expectMessageActionTakenCalled() async {
+        messagingInAppImplementation.setEventListener(eventListenerMock)
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessage(givenInlineMessage, insideOfInlineView: inlineView)
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // Tap close button on inline message
+        await onCloseActionButtonPressed(onInlineView: inlineView)
+
+        // Inline message is dismissed
+        XCTAssertFalse(isInlineViewVisible(inlineView))
+        XCTAssertNil(getInAppMessage(forView: inlineView))
+
+        // messageActionTaken called
+        XCTAssertTrue(eventListenerMock.messageActionTakenCalled)
+        XCTAssertEqual(eventListenerMock.messageActionTakenCallsCount, 1)
+    }
 }
```

Comment:
Good test case of testing when a close button pressed, we expect the event listener to be called. 

I think we need to add test functions for all of the other action button types and see if they are expected to call the event listener or not.

Reply:
By `other action button types` do you mean to test show another message button as we have custom buttons, close button and show another message button, where the first two cases have been handled and the only one I can think of is show another message. Am I understanding it right ?

Reply:
Yes, you have the correct understanding. 

Some examples of test functions I think would be valuable: 
* Given click on "Show another message" action button, expect to not call any of the event listener functions or delegate. 
* Given click on deep link button, expect to not call any of the event listener functions or delegate. 

Also, [because the `messageDismissed` event listener function is never called for inline Views](https://github.com/customerio/docs/pull/2136/files#diff-c5c6c29c2bebc6fbace8740bb04c6e9f39f2fd497eb6e5a0532f40baef7a1cd2R207), only for Modals, I think it would be valuable to write test functions where we perform actions on an inline View and make sure that we do not call the `messageDismissed` event listener function. 

Example tests: 
* Given click on close button, expect to not call `messageDismissed`. 

Maybe these test functions have already been written in a different PR, I am not sure. Since this PR is talking about tests for the global event listener, I am only going off of what I see in this PR. If you know of these tests existing already, sounds good to me!